x/subaccount/keeper/create_subaccount.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/sge-network/sge/x/subaccount/types"
+)
+
+func (m msgServer) CreateSubAccount(
+	ctx context.Context,
+	request *types.MsgCreateSubAccount,
+) (*types.MsgCreateSubAccountResponse, error) {
+	sdkContext := sdk.UnwrapSDKContext(ctx)
+	moneyToSend := sdk.NewInt(0)
+
+	err := request.ValidateBasic()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid request")
+	}
+
+	for _, balanceUnlock := range request.LockedBalances {
+		if balanceUnlock.UnlockTime.Unix() < sdkContext.BlockTime().Unix() {
+			return nil, types.ErrUnlockTokenTimeExpired
+		}
+
+		moneyToSend = moneyToSend.Add(balanceUnlock.Amount)
+	}
+
+	senderAccount, _ := sdk.AccAddressFromBech32(request.Sender)
+	subaccountOwner, _ := sdk.AccAddressFromBech32(request.SubAccountOwner)
+	if m.keeper.HasSubAccount(sdkContext, subaccountOwner) {
+		return nil, types.ErrSubaccountAlreadyExist
+	}
+
+	subaccountID := m.keeper.NextID(sdkContext)
+
+	// ALERT: If someone frontruns the account creation, will be overwritten here
+	subaccountAddress := types.NewAddressFromSubaccount(subaccountID)
```

Comment:
I'm wondering, Is creating module accounts is a good approach to store sub accounts? Can we use BasicAccounts for this purpose?

Reply:
These accounts are BaseAccounts, not module accounts and this is needed for some module account trickery happening in the SDK, please refer to how cosmwasm does contract creation, it's basically for the same purpose of ours: https://github.com/CosmWasm/wasmd/blob/main/x/wasm/keeper/keeper.go#L280